packages/components/OptimizedImage.tsx

```diff
@@ -22,12 +22,27 @@ export const OptimizedImage: React.FC<
     const sourceWidth = PixelRatio.getPixelSizeForLayoutSize(width);
     const sourceHeight = PixelRatio.getPixelSizeForLayoutSize(height);
 
+    useEffect(() => {
+      setIsError(false);
+    }, [baseSourceURI]);
+
+    useEffect(() => {
+      setIsFallbackError(false);
+    }, [fallbackURI]);
+
     if ((shouldUseFallback && !fallbackURI) || isFallbackError) {
       return (
         <View
           style={{
             width,
             height,
+            position: StyleSheet.flatten(other.style).position,
+            top: StyleSheet.flatten(other.style).top,
+            left: StyleSheet.flatten(other.style).left,
+            right: StyleSheet.flatten(other.style).right,
+            bottom: StyleSheet.flatten(other.style).bottom,
+            borderColor: StyleSheet.flatten(other.style).borderColor,
+            borderWidth: StyleSheet.flatten(other.style).borderWidth,
```

Comment:
StyleSheet.flatten is an expensive operation, you should do it only once there
Especially in a component used that much

Reply:
Fixed